core/services/vrf/v2/bhs_feeder_test.go

```diff
@@ -0,0 +1,145 @@
+	"github.com/smartcontractkit/chainlink/v2/core/services/vrf/vrftesthelpers"
+	"github.com/smartcontractkit/chainlink/v2/core/store/models"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestStartHeartbeats(t *testing.T) {
+	t.Parallel()
+	ownerKey := cltest.MustGenerateRandomKey(t)
+	uni := newVRFCoordinatorV2Universe(t, ownerKey, 2)
+
+	vrfKey := cltest.MustGenerateRandomKey(t)
+	sendEth(t, ownerKey, uni.backend, vrfKey.Address, 10)
+	gasLanePriceWei := assets.GWei(1)
+	gasLimit := 3_000_000
+
+	consumers := uni.vrfConsumers
+
+	// generate n BHS keys to make sure BHS job rotates sending keys
+	var bhsKeyAddresses []string
+	var keySpecificOverrides []toml.KeySpecific
+	var keys []interface{}
+	for i := 0; i < len(consumers); i++ {
+		bhsKey := cltest.MustGenerateRandomKey(t)
+		bhsKeyAddresses = append(bhsKeyAddresses, bhsKey.Address.String())
+		keys = append(keys, bhsKey)
+		keySpecificOverrides = append(keySpecificOverrides, toml.KeySpecific{
+			Key:          ptr(bhsKey.EIP55Address),
+			GasEstimator: toml.KeySpecificGasEstimator{PriceMax: gasLanePriceWei},
+		})
+		sendEth(t, ownerKey, uni.backend, bhsKey.Address, 10)
+	}
+	keySpecificOverrides = append(keySpecificOverrides, toml.KeySpecific{
+		// Gas lane.
+		Key:          ptr(vrfKey.EIP55Address),
+		GasEstimator: toml.KeySpecificGasEstimator{PriceMax: gasLanePriceWei},
+	})
+
+	keys = append(keys, ownerKey, vrfKey)
+
+	config, _ := heavyweight.FullTestDBV2(t, "vrfv2_needs_blockhash_store", func(c *chainlink.Config, s *chainlink.Secrets) {
+		simulatedOverrides(t, gasLanePriceWei, keySpecificOverrides...)(c, s)
+		c.EVM[0].MinIncomingConfirmations = ptr[uint32](2)
+		c.Feature.LogPoller = ptr(true)
+		c.EVM[0].FinalityDepth = ptr[uint32](2)
+		c.EVM[0].GasEstimator.LimitDefault = ptr(uint32(gasLimit))
+		c.EVM[0].LogPollInterval = models.MustNewDuration(time.Second)
+	})
+
+	heartbeatPeriod := 5 * time.Second
+	// lggr, logs := logger.TestLoggerObserved(t, zapcore.DebugLevel)
+
+	t.Run("bhs_feeder_startheartbeats_happy_path", func(tt *testing.T) {
+		// consumerContracts := uni.consumerContracts
+		// consumerContractAddresses := uni.consumerContractAddresses
+		coordinator := uni.rootContract
+		coordinatorAddress := uni.rootContractAddress
+		batchCoordinatorAddress := uni.batchCoordinatorContractAddress
+		vrfOwnerAddress := ptr(uni.vrfOwnerAddress)
+		vrfVersion := vrfcommon.V2
+
+		app := cltest.NewApplicationWithConfigV2AndKeyOnSimulatedBlockchain(t, config, uni.backend, keys...)
+		require.NoError(t, app.Start(testutils.Context(t)))
+
+		// Create VRF job.
+		vrfJobs := createVRFJobs(
+			t,
+			[][]ethkey.KeyV2{{vrfKey}},
+			app,
+			coordinator,
+			coordinatorAddress,
+			batchCoordinatorAddress,
+			uni.coordinatorV2UniverseCommon,
+			vrfOwnerAddress,
+			vrfVersion,
+			false,
+			gasLanePriceWei)
+		_ = vrfJobs[0].VRFSpec.PublicKey.MustHash()
+
+		var (
+			v2CoordinatorAddress     string
+			v2PlusCoordinatorAddress string
+		)
```

Comment:
Do we need to run VRF for this test that just tests heartbeats?

Reply:
Removed the VRF job. Wanted to test with a proper a VRF setup. Can remove it, since its not directly relevant to this test.